DeviceManager/TemplateHandler.py

```diff
@@ -182,12 +182,22 @@ def update_template(req, template_id):
         updated, json_payload = parse_payload(req, template_schema)
         old.label = updated['label']
 
-        for attr in old.attrs:
-            db.session.delete(attr)
-        db.session.flush()
-        for attr in json_payload['attrs']:
-            mapped = DeviceAttr(template=old, **attr)
-            db.session.add(mapped)
+        new = json_payload['attrs']
+        for a in old.attrs:
+            found = False
+            for idx, b in enumerate(new):
+                if (a.label == b['label']) and (a.type == b['type']):
+                    found = True
+                    a.value_type = b.get('value_type', None)
+                    a.static_value = b.get('static_value', None)
+                    new.pop(idx)
+                    break
+            if not found:
```

Comment:
tip: for's else statement is great for this situation, to eliminate 'found' flags

Reply:
Yes, I really had never thought of that.
I found some [discussion](https://mail.python.org/pipermail/python-ideas/2009-October/006155.html) on the topic - actually from a little while now.

Fun fact - I literally uttered the words when we spoke:
> mmm, isn't this `else` clause mis-indented somehow?

I guess the author does know his audience. :thinking: